product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/vfs/VFSClient.java

```diff
@@ -0,0 +1,222 @@
+     * @param targetFilePath target path of the remote file
+     * @throws IOException if file upload fails
+     */
+    public void ftpFileUpload(String sourceFilePath, String targetFilePath) throws IOException {
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+        File localFile = new File(sourceFilePath);
+
+        try (InputStream inputStream = new FileInputStream(localFile)) {
+            log.info("Start uploading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.storeFile(targetFilePath, inputStream)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("IO Exception occurred while uploading file ", ex);
+            }
+        } catch (FileNotFoundException ex) {
+            throw new IOException("File Not Found Exception occurred ", ex);
+        }
+    }
+
+    /**
+     * Function to download a file from an FTP Server
+     *
+     * @param sourceFilePath source file path
+     * @param targetFilePath target path
+     * @throws IOException if file download fails
+     */
+    public void ftpFileDownload(String sourceFilePath, File targetFilePath) throws IOException{
+        ftpClient.setFileType(FTP.BINARY_FILE_TYPE);
+
+        try (OutputStream outputStream1 = new BufferedOutputStream(new FileOutputStream(targetFilePath))) {
+            log.info("Start downloading file " + sourceFilePath + " to " + targetFilePath);
+
+            try {
+                if (ftpClient.retrieveFile(sourceFilePath, outputStream1)) {
+                    log.info("File has been uploaded to " + targetFilePath + " successfully.");
+                }
+            } catch (IOException ex) {
+                throw new IOException("IO Exception occurred while downloading file ", ex);
+            }
+        } catch (FileNotFoundException ex) {
+            throw new IOException("File Not Found Exception occurred ", ex);
+        }
+    }
+
+    /**
+     * Function to list the set of directories on an FTP server
+     *
+     * @param filePath file path to list directories
+     * @return list of directories
+     * @throws IOException if listing set of directories fails
+     */
+    public FTPFile[] listDirectories(String filePath) throws IOException {
+        FTPFile[] directoryList;
+        try {
+            directoryList = ftpClient.listDirectories(filePath);
+        } catch (IOException ex){
+            throw new IOException("IO Exception occurred while listing directories", ex);
+        }
+        return directoryList;
+    }
+
+    /**
+     * Function to list files of a directory on an FTP server
+     *
+     * @param filePath file path to list files
+     * @return list of files
+     * @throws IOException if listing set of files fails
+     */
+    public List<FTPFile> listFiles(String filePath) throws IOException {
```

Comment:
We return an **array** of FTPFile in listDirectories. Why do we return a **List** in this? Shal we make the 2 methods consistent?